Ginger/Ginger/UserControlsLib/UCTreeView/SingleItemTreeViewSelectionPage.xaml.cs

```diff
@@ -178,5 +185,18 @@ private void Tree_ItemDoubleClick(object sender, EventArgs e)
                 }
             }
         }
+        private void Tree_ItemSelected(object sender, EventArgs e)
+        {            
+            mSelectedItems = new List<object>();
+            ITreeViewItem itvItem = xTreeView.Tree.CurrentSelectedTreeViewItem;
+            if(itvItem.NodeObject() == null)
+                mSelectedItems.Add(itvItem);
+            else
+                mSelectedItems.Add(itvItem.NodeObject());
+            if (SelectCurrentItem())
+            {
+                OnSelectionDone(new SelectionTreeEventArgs(mSelectedItems));
+            }            
+        }
     }
```

Comment:
can you please elaborate what this code is for?

Reply:
This is adding the selecting item to the selectionlist in case node object of selected item is null.
I need the path of the selected node from the calling class. Earlier we were  updating the list only on double click , i added it for select event

Reply:
But you want to select only single folder, no? so why list? the actual selection is done with the "Select" button which takes the current selectedItem, i guess i'm missing something...

Reply:
We do not have select button here. as we are not opening it as window. So just implemented the same we had in select click. Also, yes we only need single item selected, but then we will need to define another selection argument which will return object instead of List<object>. So i kept it as list and outside reading the first element from the list